### ISSUE TITLE
Separate out logic in CoC creation tool

### ISSUE THREAD

**Issue by trickyturtle** [trickyturtle/RPGscripts#1]
CoCCharGen is...a mess.  Too many functions, separate them out.

Design Problem:
Should I 
A) just leave all the randomization functions as they are (incredibly long switch statements)
 
This is easiest obviously, and since this project doesn't really need to worry about scaling, there is no reason to worry about the performance hit from doing it this way.

B) convert them into yaml.

Obviously this would involve a hash of the text, and then the action(s) to take if any (probably using some kind of messaging system).  This would be a pain, but also makes it easier to convert this into a different language later if I want to.  Probably go this route just for experience

**Comment by trickyturtle** [comment]
Also, if somebody (joe) is reading this, I'm saying we should separate the functions into different ruby classes because right now CoCCharGen is too big and complicated.  In general, you want to try to keep classes less than 100 lines, so they are easier to read.

The common example is if you were building a bike, you don't want to make the bike such that every part of it is welded on (so you can't make a new seat, change the wheel style, make it a single speed, etc).  You want to have lots of smaller parts that you can easily exchange for some other part.  It makes it easier to diagnose problems and to make changes down the road if you decide you want to.

For example, the birth function (which just vaguely creates a backstory for the character) should just be it's own class (I think I left it in CoCCharGen, but if not it hasn't really changed in its new function), because it isn't intrinsically tied to CoC, in theory anything could use it.  Well...actually in it's current for it is intrinsically tied to CoC because it uses it's stats, but there isn't really any reason for it to be doing that (one of the "issues" in the issue tracker is that it should be converted into yaml, to make the birth function more generic).

tl;dr - you want classes to be short and abstract, often you can achieve this by breaking apart long classes and making their function just generic classes.
